crates/sui-framework/sources/epoch_time_lock.move

```diff
@@ -4,6 +4,7 @@
 module sui::epoch_time_lock {
     use sui::tx_context::{Self, TxContext};
     friend sui::sui_system;
+    friend sui::staking_pool;
```

Comment:
Needed? Doesn't seem to be used anywhere so I'm guessing this can be removed.

Reply:
We actually need it here so `staking_pool::join` can call `destroy_unchecked`which is a friend fun.